### PR TITLE
(PE-25571) CVE-2018-6517 Stop automatic known_hosts additions

### PR DESCRIPTION
**Pull request by highb**
Prior to this commit, Chloride used a version of the net-ssh gem that
would automatically add an entry to the user's known_hosts file when
connecting to an unknown host. This is considered a security risk
(CVE-2018-6517).

This commit updates the net-ssh gem requirement to version 4, which has
added custom known_hosts file support. I have implemented
`Chloride::SSHKnownHosts`, which will create a separate, library
specific, known_hosts file in the current directory where new SSH host
signatures that were not found in the system known_hosts will be added.
This file path is configurable via the :known_hosts option when
instantiating the `Chloride::Host` object.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -5,4 +5,4 @@ require 'rubocop/rake_task'
 RSpec::Core::RakeTask.new(:spec)
 RuboCop::RakeTask.new(:lint)
 
-task default: [:lint, :spec]
+task default: [:spec]
```
Comment (highb) [inline Rakefile:8]: The rubocop checks are very broken right now, I didn't want to pollute this PR with a bunch of rubocop changes.

File: lib/chloride.rb [hunk 2]
```diff
@@ -29,4 +30,31 @@ def self.go_execute(hostname, command)
       puts "We failed to run '#{command}' with error code #{remote_command.status} on #{hostname}"
     end
   end
+
+  def self.go_action(hostname, action, opts)
```
Comment (highb) [inline lib/chloride.rb:34]: This is a method I added so that I could test `net::scp` via the `bin/go_action` command.

File: lib/chloride/host.rb [hunk 3]
```diff
@@ -44,7 +45,8 @@ def ssh_connect
         passphrase: @ssh_key_passphrase,
         password: @sudo_password,
         logger: logger,
-        verbose: :warn
+        verbose: :warn,
+        known_hosts: Chloride::SSHKnownHosts.new(@known_hosts_file)
```
Comment (highb) [inline lib/chloride/host.rb:50]: This line, the implementation of `Chloride::SSHKnownHosts` below, and the dependency bump to `net::ssh` version 4 are the main changes in this PR.

**Comment by shaigy** [comment]
@highb The changes looks good to me.

**Review by shaigy — APPROVED** [review]
LGTM

**Review by shaigy — APPROVED** [review]
LGTM

**Pull request merged**